File: templates/problems/17_math_geometry/05_plus_one/go/plus_one_test.go
package plus_one

import (
	"reflect"
	"testing"
)

func TestPlusOne(t *testing.T) {
	tests := []struct {
		name     string
		digits   []int
		expected []int
	}{
		{"no carry", []int{1, 2, 3}, []int{1, 2, 4}},
		{"single carry", []int{4, 3, 2, 9}, []int{4, 3, 3, 0}},
		{"all nines", []int{9, 9, 9}, []int{1, 0, 0, 0}},
		{"single digit", []int{0}, []int{1}},
		{"single nine", []int{9}, []int{1, 0}},
		{"large number", []int{8, 9, 9, 9}, []int{9, 0, 0, 0}},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := plusOne(tt.digits)
			if !reflect.DeepEqual(got, tt.expected) {
				t.Errorf("plusOne(%v) = %v, want %v", tt.digits, got, tt.expected)
			}
		})
	}
}

File: templates/problems/17_math_geometry/05_plus_one/typescript/plusOne.test.ts
import { describe, it, expect } from 'vitest';
import { plusOne } from './plusOne';

describe('PlusOne', () => {
    it('no carry', () => {
        expect(plusOne([1, 2, 3])).toEqual([1, 2, 4]);
    });

    it('single carry', () => {
        expect(plusOne([4, 3, 2, 9])).toEqual([4, 3, 3, 0]);
    });

    it('all nines', () => {
        expect(plusOne([9, 9, 9])).toEqual([1, 0, 0, 0]);
    });

    it('single digit', () => {
        expect(plusOne([0])).toEqual([1]);
    });

    it('single nine', () => {
        expect(plusOne([9])).toEqual([1, 0]);
    });

    it('large number', () => {
        expect(plusOne([8, 9, 9, 9])).toEqual([9, 0, 0, 0]);
    });
});
